packages/order-experience-api/src/api/confirmAndCreatePayment.ts

```diff
@@ -39,6 +42,7 @@ export class ConfirmAndCreatePayment extends AbstractController<
     } = req
 
     const order = await confirmOrder({ orderId, user })
+    checkLastValidPurchaseDateTime(order.lastValidPurchaseDateTime)
```

Comment:
Should this be before we do confirm to order? 

If you need order data, you can fetch it with getOrderAdmin function

Reply:
As discussed in the daily, I shall change this.